identity/src/main/java/com/stripe/android/identity/ui/IdentityTheme.kt

```diff
@@ -145,6 +179,46 @@ private inline val Resources.Theme.key: Any?
         return null
     }
 
+
+@Composable
+private fun detectHostThemeType(): HostThemeType {
+    val context = LocalContext.current
+
+    val isMaterialTheme = remember {
+        runCatching {
+            context.obtainStyledAttributes(MaterialR.styleable.ThemeAdapterMaterialTheme).use { ta ->
+                ta.hasValue(MaterialR.styleable.ThemeAdapterMaterialTheme_isMaterialTheme)
+            }
+        }.getOrDefault(false)
+    }
+
+    val isMaterial3Theme = remember {
+        runCatching {
+            context.obtainStyledAttributes(Material3R.styleable.ThemeAdapterMaterial3Theme).use { ta ->
+                ta.hasValue(Material3R.styleable.ThemeAdapterMaterial3Theme_isMaterial3Theme)
+            }
+        }.getOrDefault(false)
+    }
+
+    return when {
+        isMaterialTheme -> HostThemeType.MaterialComponents
+        isMaterial3Theme -> HostThemeType.Material3
+        else -> HostThemeType.AppCompat
+    }
+}
+
+private enum class HostThemeType {
+    MaterialComponents,
+    Material3,
+    AppCompat
+}
```

Comment:
Extracted this from payments. Cannot move to a shared module (`stripe-ui-core`) since it requires mdc / material theming related dependencies that other SDKs might not use.